src/App.tsx

```diff
@@ -1,223 +1,210 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error';
+import { Navigation } from './components/Navigation';
+import { Filter } from './types/Filter';
+import { NewTodo } from './components/NewTodo';
+import { TodoList } from './components/TodoList';
+import { LoaderContext } from './components/Context/LoaderContext';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [error, setError] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const { setTodoOnLoad, setTodosOnLoad } = useContext(LoaderContext);
+
+  const loadTodos = async () => {
+    if (!user) {
+      return;
+    }
+
+    const todosFromServer = await getTodos(user.id);
+
+    setTodos(todosFromServer);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    loadTodos();
+  }, [user]);
+
+  const filterTodos = useCallback((filterBy: Filter) => (
+    todos.filter(todo => {
+      switch (filterBy) {
+        case Filter.ALL:
+          return todo;
+
+        case Filter.ACTIVE:
+          return !todo.completed;
+
+        case Filter.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    })), [todos, filter]);
+
+  const filteredTodos = filterTodos(filter);
+
+  const completedTodos = filterTodos(Filter.COMPLETED);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const removeError = useCallback(() => {
+    setError('');
+  }, []);
+
+  const handleError = useCallback((errorText: string) => {
+    setError(errorText);
+    setTimeout(removeError, 3000);
+  }, []);
+
```

Comment:
Maybe I just didn’t find it, but it seems to me that you don’t stop the timer anywhere ... I think this needs to be fixed)

Reply:
I dont set timer (it is setInterval function, isn't it?) , i set timeout) error will disappear after 3000 ms

Reply:
Maybe I'm wrong ... It seemed to me that when the settimeout was called, a timer was started that needed to be stopped.